super-ddf.c

```diff
@@ -3035,7 +3042,8 @@ static int __write_ddf_structure(struct dl *d, struct ddf_super *ddf, __u8 type)
 	header->openflag = 0;
 	header->crc = calc_crc(header, 512);
 
-	lseek64(fd, sector<<9, 0);
+	if (lseek64(fd, sector<<9, 0) < 0)
+		ret = 0;
```

Comment:
I think that since we couldn't reposition offset, we shouldn't try to write header below, so here should be return 0;